src/helpers/createPdfAndSave.js

```diff
@@ -0,0 +1,154 @@
+import { PDFDocument } from 'pdf-lib'
+
+import { models } from 'cozy-client'
+
+import { CONTACTS_DOCTYPE, FILES_DOCTYPE } from 'src/doctypes'
+import { addFileToPdf } from 'src/utils/addFileToPdf'
+import { buildFilename } from 'src/helpers/buildFilename'
+
+const {
+  contact: { getFullname },
+  file: { uploadFileWithConflictStrategy }
+} = models
+
+/**
+ * @typedef {object} SaveFileParam
+ * @property {CozyClient} client
+ * @property {DocumentCollection} fileCollection
+ * @property {ArrayBuffer} pdfBytes
+ * @property {string} contactId
+ * @property {{ name: string, metadata: object, dirId: string }} opts
+ */
+
+/**
+ * @typedef {object} createAndSavePdfParam
+ * @property {{ data: object[], metadata: object }} formData
+ * @property {Qualification} qualification
+ * @property {Paper} currentDefinition
+ * @property {string} appFolderID
+ * @property {CozyClient} client
+ * @property {{t: Function, f: Function, scannerT: Function}} i18n
+ */
+
+/**
+ * @param {{multipage: boolean, page: string}} fileMetadata
+ * @param {boolean} isMultipage
+ * @returns {object}
+ */
+const sanitizeFileMetadata = (fileMetadata, isMultipage) => {
+  if (isMultipage || !fileMetadata.page) return {}
+
+  // eslint-disable-next-line no-unused-vars
+  const { multipage, ...newFileMetadata } = fileMetadata
+
+  return newFileMetadata
+}
+
+/**
+ * @param {SaveFileParam} param
+ * @returns {Promise<string>} Filename
+ */
+const saveFile = async ({
+  client,
+  fileCollection,
+  pdfBytes,
+  contacts,
+  opts
+}) => {
+  const { name, metadata, dirId } = opts
+  const { data: fileCreated } = await uploadFileWithConflictStrategy(
+    client,
+    pdfBytes,
+    {
+      name,
+      contentType: 'application/pdf',
+      metadata,
+      dirId,
+      conflictStrategy: 'rename'
+    }
+  )
+
+  // The user is referenced as a contact on the new Paper
+  const references = contacts.map(contact => ({
+    _id: contact._id,
+    _type: CONTACTS_DOCTYPE
+  }))
+  await fileCollection.addReferencedBy(fileCreated, references)
+
+  return fileCreated._id
+}
+
+/**
+ * Convert image & pdf file to pdf & save it
+ *
+ * @param {createAndSavePdfParam} param
+ * @returns {Promise<{ fileId: string, themeLabel: string }[]>} Return array of object with file id & theme label to find its location
+ */
+export const createPdfAndSave = async ({
+  formData,
+  qualification,
+  currentDefinition,
+  appFolderID,
+  client,
+  i18n
+}) => {
+  const { t, f, scannerT } = i18n
+  const { data, metadata, contacts } = { ...formData }
+  const fileCollection = client.collection(FILES_DOCTYPE)
+  const isMultiPage = data.some(({ fileMetadata }) => fileMetadata.multipage)
+  const { featureDate, label } = currentDefinition
+  const date = metadata[featureDate] && f(metadata[featureDate], 'YYYY.MM.DD')
+  let fileCreated = []
+
+  // Created first page of pdf
```

Comment:
This is not the first page of pdf. This is just a new empty pdf document.

Maybe we should avoid using the same term (page) for bot pdf document and `Mes papiers`'s form pages. Those are ambiguous.

Reply:
Pdf pages are created inside of `addFileToPdf`